fullstack-cert/typescript-projects/digital-pet-game/src/App.tsx

```diff
@@ -0,0 +1,301 @@
+import { useState, useEffect, useRef } from "react";
+import "./App.css";
+import * as catFacts from "./assets/cat-facts.json";
+
+interface StatBarProps {
+  label: string;
+  value: number;
+  icon: string;
+}
+
+function getStatColorClass(value: number): string {
+  if (value >= 70) return "high";
+  if (value >= 40) return "medium";
+  return "low";
+}
+
+function StatBar({ label, value, icon }: StatBarProps) {
+  const colorClass = getStatColorClass(value);
+
+  return (
+    <div className="stat-bar">
+      <div className="stat-header">
+        <div className="stat-label">
+          <span className="stat-icon">{icon}</span>
+          <span className="stat-name">{label}</span>
+        </div>
+        <span className="stat-value">{Math.round(value)}%</span>
+      </div>
+
+      <div className="stat-progress">
+        <div
+          className={`stat-fill ${colorClass}`}
+          style={{ width: `${value}%` }}
+        />
+      </div>
+    </div>
+  );
+}
+
+enum PetAction {
+  NONE,
+  EAT,
+  PLAY,
+  SLEEP,
+}
+
+interface Pet {
+  name: string;
+  happiness: number;
+  hunger: number;
+  energy: number;
+  species: string;
+}
+
+enum PetMood {
+  Happy,
+  Excited,
+  Content,
+  Sad,
+  Tired,
+  Sick,
+  Hungry,
+}
+
+export function calculatePetMood(pt: Pet): PetMood {
+  const { hunger, happiness, energy } = pt;
+
+  if (hunger > 70) return PetMood.Hungry;
+  if (energy < 30) return PetMood.Tired;
+  if (happiness < 30) return PetMood.Sad;
+  if (happiness > 80 && energy > 70) return PetMood.Excited;
+  if (happiness > 60) return PetMood.Happy;
+
+  return PetMood.Content;
+}
+
+const moodEmojiMap: Record<PetMood, string> = {
+  [PetMood.Happy]: "😺", // grinning cat
+  [PetMood.Excited]: "😻", // heart-eyes cat
+  [PetMood.Content]: "😸", // smiling cat
+  [PetMood.Sad]: "😿", // crying cat
+  [PetMood.Tired]: "😽", // kissing cat (closest to sleepy)
+  [PetMood.Sick]: "🙀", // weary cat (looks unwell)
+  [PetMood.Hungry]: "😹", // joy cat (closest to hungry/funny)
+};
+
+const STORAGE_KEY = "pet";
+
+interface usePetProps {
+  isGameStarted: boolean;
+  setGameStarted: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+function usePet({ isGameStarted, setGameStarted }: usePetProps) {
+  let [pet, setPet] = useState<Pet>({
+    name: "",
+    happiness: 100,
+    hunger: 0,
+    energy: 100,
+    species: "Cat",
+  });
+
+  useEffect(() => {
+    console.log("PLEASE BORTHER!");
+    if (!isGameStarted) return;
+    console.log("Excellent, Borther.");
```

Comment:
I am assuming these console.log statements were for testing? if so, we can remove them here